tensorflow_addons/optimizers/proximal_adagrad.py

```diff
@@ -0,0 +1,142 @@
+
+from typing import Union, Callable
+from typeguard import typechecked
+
+
+@tf.keras.utils.register_keras_serializable(package="Addons")
+class ProximalAdagrad(tf.keras.optimizers.Optimizer):
+    """Optimizer that implements the Proximal Adagrad algorithm.
+
+    References:
+        - [Efficient Learning using Forward-Backward Splitting](
+          http://papers.nips.cc/paper/3793-efficient-learning-using-forward-backward-splitting.pdf).
+    """
+
+    @typechecked
+    def __init__(
+        self,
+        learning_rate: Union[FloatTensorLike, Callable] = 0.001,
+        initial_accumulator_value: float = 0.1,
+        l1_regularization_strength: float = 0.0,
+        l2_regularization_strength: float = 0.0,
+        name: str = "ProximalAdagrad",
+        **kwargs
+    ):
+        """Construct a new Proximal Adagrad optimizer.
+
+        Args:
+            learning_rate: A Tensor or a floating point value, or a schedule
+                that is a `tf.keras.optimizers.schedules.LearningRateSchedule`.
+                The learning rate.
+            initial_accumulator_value: A floating point value.
+                Starting value for the accumulators, must be positive.
+            l1_regularization_strength: A floating point value.
+                The l1 regularization term, must be greater than or
+                equal to zero.
+            l2_regularization_strength: A floating point value.
+                The l2 regularization term, must be greater than or
+                equal to zero.
+            name: Optional name for the operations created when applying
+                gradients. Defaults to "ProximalAdagrad".
+            **kwargs: keyword arguments. Allowed to be {`clipnorm`,
+                `clipvalue`, `lr`, `decay`}. `clipnorm` is clip gradients
+                by norm; `clipvalue` is clip gradients by value, `decay` is
+                included for backward compatibility to allow time inverse
+                decay of learning rate. `lr` is included for backward
+                compatibility, recommended to use `learning_rate` instead.
+        Raises:
+            ValueError: If the `initial_accumulator_value`,
+                `l1_regularization_strength` or `l2_regularization_strength`
+                is invalid.
+        """
+        if initial_accumulator_value < 0.0:
+            raise ValueError("`initial_accumulator_value` must be non-negative.")
+        if l1_regularization_strength < 0.0:
+            raise ValueError("`l1_regularization_strength` must be non-negative.")
+        if l2_regularization_strength < 0.0:
+            raise ValueError("`l2_regularization_strength` must be non-negative.")
+        super().__init__(name, **kwargs)
+        self._set_hyper("learning_rate", kwargs.get("lr", learning_rate))
+        self.l1_regularization_strength = l1_regularization_strength
+        self.l2_regularization_strength = l2_regularization_strength
```

Comment:
How about creating l1_regularxx and l2_regularxx as hyper-parameter?

Reply:
Will do. Just copy paste the name from `compat.v1`. Maybe `l1` and `l2` are more suitable?

https://www.tensorflow.org/api_docs/python/tf/keras/regularizers/L1L2

Reply:
I agree, `l1` and `l2` are more concise :-)

Reply:
Tzu-Wei, perhaps we could use `self._set_hyper` for `l1` and `l2`, what do you think?

Reply:
of course. let me update it real quick :-)

Reply:
Seems that `_set_hyper`/`_get_hyper` supports more types of inputs.